server/sonar-server/src/main/java/org/sonar/server/search/ws/BaseMapping.java

```diff
@@ -27,14 +27,15 @@
 import javax.annotation.Nullable;
 import org.sonar.api.server.ServerSide;
 import org.sonar.api.utils.text.JsonWriter;
-import org.sonar.server.search.BaseDoc;
+import org.sonar.server.es.BaseDoc;
 import org.sonar.server.search.IndexUtils;
 import org.sonar.server.search.QueryContext;
 
 /**
- * Mapping of search documents (see {@link org.sonar.server.search.BaseDoc}) to WS JSON responses
+ * Mapping of search documents (see {@link BaseDoc}) to WS JSON responses
  */
 @ServerSide
+@Deprecated
 public abstract class BaseMapping<DOC extends BaseDoc, CTX> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)